### ISSUE TITLE
Can't use lightningconductor.net

### ISSUE THREAD

**Issue by mmvphil** [ElementsProject/lightning#2059]
### Issue and Steps to Reproduce

Tried their service .. no go moving LN >> BTC
I knew there was an issue with return chans but I can't even use it for liquidity provider?
Is there a fix for this?

**Comment by jb55** [comment]
c-lightning has no plans on supporting return channels in the way the lightning conductors implements them as it requires two channels to be opened between nodes. The only reliable way to gain inbound capacity with clightning at this time is to spend some of your outbound capacity, or to convince others to open a channel with you first.

**Comment by cdecker** [comment]
So let me quickly understand how lightningconductor works: you open a connection (and channel) with them, and if you send them coins they'll open a channel with you?

How about you open a connection, but don't open a channel over it? Then it should still be able to fund a channel with you as expected.

**Comment by jb55** [comment]
@cdecker they only open a channel with you once you've opened a channel with them

**Comment by jb55** [comment]
@cdecker how so? it's a way of getting inbound capacity without having to arbitrarily spend any of your outbound capacity. Let's say I spin up a new node and I want to start receiving payments. I could ask every single person to connect to me but it would be easier if I could quickly bootstrap onto the network with inbound capacity, afaict that's the purpose of lightning conductor.

It seems to be a popular hack at the moment, yalls provides a similar service as well.

**Comment by cdecker** [comment]
No, what I'm saying is that if the goal is to get inbound capacity, why do I have to open an outgoing channel first? I could connect to anybody else, and get only my inbound capacity form this service.

**Comment by mmvphil** [comment]
@jb55 "I could ask every single person to connect to me but it would be easier "
mate forgive my ignorance but if a person connects to me because i have no inbound capacity left does that not imply an on chain tx?  I'm confused.. cause if we have to do a tx for each inbound payment??

From a merchants (95% inbound) perspective what is the best practice here?
Open up a giant channel where I know i can get payment back and withdraw once out of capacity?

**Comment by jb55** [comment]
Christian Decker <notifications@github.com> writes:

> No, what I'm saying is that if the goal is to get inbound capacity,
> why do I have to open an outgoing channel first? I could connect to
> anybody else, and get only my inbound capacity form this service.

I think the issue is that for a public service that offers channel
openings, they are taking on some risk due to channel closings + losing
money to fees, so they require the other party to put some skin in the
game as well.

**Comment by jb55** [comment]
mmvphil <notifications@github.com> writes:

> Spending capacity
>
> @jb55 "The only reliable way to gain inbound capacity with clightning at this time is to spend some of your outbound capacity, or to convince others to open a channel with you first."
>
> Is that not what https://lightningconductor.net/invoice would do ? LN -> BTC ??

It looks like it, but it's kind of a strange way to get inbound
capacity. You would lose a bunch to onchain and service fees, and it
looks like you have to trust them to send the BTC back?

**Comment by jb55** [comment]
mmvphil <notifications@github.com> writes:

> From a merchants (95% inbound) perspective what is the best practice here?
> Open up a giant channel where I know i can get payment back and withdraw once out of capacity?

As a merchant you would want to have enough inbound capacity to handle
payments from people how haven't directly connnected. In theory if you
are a busy shop, it would be smart for well connected nodes to connect
to you to collect fees, so the market should solve these problems in the
long run.

For now you just have to ask the user to connect to your node to make
the payment, or ask others by letting them know you are opening shop and
are expecting lots of business.

In either case, it helps that you have a healthy outbound channel to
another node that needs inbound capacity to improve the health of the
network.

Eventually things like autopilot should be able to do this for you, as
it would get too complex/tiresome for us humans to manage manually.

Let me know if this makes sense.

Cheers,

**Comment by mmvphil** [comment]
@jb55 .. hmm kinda makes sense but "just have to ask the user to connect to your node"
Does that require them to FUND my channel as well or just connect? Cause if they need to fund it then that is an onchain Tx and we are no farther ahead vis a vis scale ability.

**Comment by jb55** [comment]
@mmvphil afaik connecting doesn't do much other then allow you to trade gossip messages? Everything I mentioned above was in context of opening channels (onchain txs).

> and we are no farther ahead vis a vis scale ability

hmmm not quite, they can still use your channel to make payments through you if you have outgoing capacity. This is really just a question of bootstrapping initial capacity between nodes which should improve over time with autopilot and market mechanisms.

**Comment by mmvphil** [comment]
This is the entire reason we need chan liquidity provider like lightningconductor however they don't seem very responsive as to why its not working from c-lightning.   So I have built and app and it's just sitting there waiting for launch as in our test runs via 'bitcoin lightning wallet' we have issues accepting payments so entire project is on stall mode.  I guess we can go ahead and let them fund our channel but not sure people will want to do that.. I guess we will see. :)
